src/GeneratorV2/Process/Processor.cs

```diff
@@ -871,22 +873,22 @@ public void WritePrologue(IndentedTextWriter writer)
             }
         }
 
-        public class GenAndCreateOverloader : IOverloader
+        public class GenCreateAndDeleteOverloader : IOverloader
         {
             public bool TryGenerateOverloads(Overload overload, [NotNullWhen(true)] out List<Overload>? newOverloads)
             {
                 // Here we assume that the last parameter is the pointer parameter.
                 var pointerParameter = overload.InputParameters.LastOrDefault();
                 var nativeName = overload.NativeFunction.FunctionName;
-                if ((!nativeName.StartsWith("Create") && !nativeName.StartsWith("Gen")) ||
+                if ((!nativeName.StartsWith("Create") && !nativeName.StartsWith("Gen") && !nativeName.StartsWith("Delete")) ||
                     !nativeName.EndsWith("s") || pointerParameter == null ||
                     pointerParameter.Type is not CSPointer pointerParameterType ||
                     pointerParameter.Length == null || pointerParameter.Length is not ParameterReference handleLength)
                 {
                     newOverloads = null;
                     return false;
                 }
-                var newNativeName = nativeName[..^1];
+                var newName = nativeName[..^1];
```

Comment:
This should add a `NameMangler.RemoveEnd("s")` to make sure we are actually removing the right thing.

Reply:
We already check that the name ends with an "s" in the above if statement

Reply:
Yeah ok, we might not catch extension methods using that though. But idk if there are any Gen, Create or Delete extension methods